File: src/main/java/musiccube/entities/Instrument.java
package musiccube.entities;

import com.fasterxml.jackson.annotation.JsonInclude;

import javax.persistence.*;

@Entity
@JsonInclude(JsonInclude.Include.NON_NULL)
@Table(name = "instrument")
public class Instrument {

    @Id
    @GeneratedValue
    @Column
    private int id;

    @Column(unique = true)
    private String instrumentName;

    @Column(length = 1024)
    private String aboutInstrument;

    @ManyToOne
    @JoinColumn(referencedColumnName = "id")
    private InstrumentType instrumentType;

    public Instrument() {}

    public Instrument(String instrumentName, InstrumentType instrumentType) {
        this.instrumentName = instrumentName;
        this.instrumentType = instrumentType;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getInstrumentName() {
        return instrumentName;
    }

    public void setInstrumentName(String instrumentName) {
        this.instrumentName = instrumentName;
    }

    public InstrumentType getInstrumentType() {
        return instrumentType;
    }

    public void setInstrumentType(InstrumentType instrumentType) {
        this.instrumentType = instrumentType;
    }

    public String getAboutInstrument() {
        return aboutInstrument;
    }

    public void setAboutInstrument(String aboutInstrument) {
        this.aboutInstrument = aboutInstrument;
    }
}

File: Frontend/src/app/Class/Instrument.ts
import { InstrumentType } from './InstrumentType';

export class Instrument {

  id: number;
  instrumentName: string;
  instrumentType: InstrumentType;
  aboutInstrument: string;

  constructor(obj?: any) {
    this.id = (obj && obj.id) || 0;
    this.instrumentName = (obj && obj.instrumentName) || '';
    this.instrumentType = (obj && obj.instrumentType) || '';
    this.aboutInstrument = (obj && obj.aboutInstrument) ||
      'No description was provided for this instrument. If you think, you could write some, please contact us.';
  }

  // Getters

  getId(): number {
    return this.id;
  }

  getInstrumentName(): string {
    return this.instrumentName;
  }

  getInstrumentType(): InstrumentType {
    return this.instrumentType;
  }

  getAboutInstrument() {
    return this.aboutInstrument;
  }

  // Setters

  setInstrumentName(instrumentName: string) {
    this.instrumentName = instrumentName;
  }

  setInstrumentType(instrumentType: InstrumentType) {
    this.instrumentType = instrumentType;
  }

  setAboutInstrument(aboutInstrument: string) {
    this.aboutInstrument = aboutInstrument;
  }
}
